### DISCUSSION TITLE
JS is weird

### DISCUSSION THREAD

**Discussion by RedMan13** [PenguinMod/PenguinMod-Home#82] [General]
for example, 
```js
Scratch.extensions.register({
    getInfo: () => ({
        id: 'joeballs',
        name: 'just a silly little guy',
        blocks: [
            {
                opcode: 'bllck',
                blockType: Scratch.BlockType.COMMAND,
                text: 'doopooopoo somthin'
            }
        ]
    }),

    bllck() {
        throw new Error('haah skipknossooo')
    }
})
``` 
is a completely valid extension

**Comment by Steve0Greatness** [comment]
I mean, yeah. But there's nothing weird in there(other than the strings).

**Reply by RedMan13** [reply]
> I mean, yeah. But there's nothing weird in there(other than the strings).

this is a comment on how you dont need a class to create a class instance, as a class instance is only an object with a custom prototype
also because javascript never specifies for prototype, treating both the prototype object and main object as the same key space, you can just make an object with the necessary function keys instead of requiring a whole class
